pinot-plugins/pinot-input-format/pinot-thrift/src/test/java/org/apache/pinot/plugin/inputformat/thrift/ThriftRecordExtractorTest.java

```diff
@@ -0,0 +1,220 @@
+    bufferedOut.close();
+  }
+
+  private Map<String, Object> createRecord1() {
+    Map<String, Object> record = new HashMap<>();
+    record.put(STRING_FIELD, "hello");
+    record.put(INT_FIELD, 10);
+    record.put(LONG_FIELD, 1000L);
+    record.put(DOUBLE_FIELD, 1.0);
+    record.put(BOOL_FIELD, "false");
+    record.put(ENUM_FIELD, TestEnum.DELTA.toString());
+    record.put(NESTED_STRUCT_FIELD, createNestedMap(NESTED_STRING_FIELD, "ice cream", NESTED_INT_FIELD, 5));
+    record.put(SIMPLE_LIST, Arrays.asList("aaa", "bbb", "ccc"));
+    record.put(COMPLEX_LIST,
+        Arrays.asList(
+            createNestedMap(NESTED_STRING_FIELD, "hows", NESTED_INT_FIELD, 10),
+            createNestedMap(NESTED_STRING_FIELD, "it", NESTED_INT_FIELD, 20),
+            createNestedMap(NESTED_STRING_FIELD, "going", NESTED_INT_FIELD, 30)
+        )
+    );
+    record.put(SIMPLE_MAP, createNestedMap("Tuesday", 3, "Wednesday", 4));
+    record.put(
+        COMPLEX_MAP,
+        createNestedMap(
+            "fruit1", createNestedMap(NESTED_STRING_FIELD, "apple", NESTED_INT_FIELD, 1),
+            "fruit2", createNestedMap(NESTED_STRING_FIELD, "orange", NESTED_INT_FIELD, 2)
+        )
+    );
+    return record;
+  }
+
+  private Map<String, Object> createRecord2() {
+    Map<String, Object> record = new HashMap<>();
+    record.put(STRING_FIELD, "world");
+    record.put(INT_FIELD, 20);
+    record.put(LONG_FIELD, 2000L);
+    record.put(DOUBLE_FIELD, 2.0);
+    record.put(BOOL_FIELD, "false");
+    record.put(ENUM_FIELD, TestEnum.GAMMA.toString());
+    record.put(NESTED_STRUCT_FIELD, createNestedMap(NESTED_STRING_FIELD, "ice cream", NESTED_INT_FIELD, 5));
+    record.put(SIMPLE_LIST, Arrays.asList("aaa", "bbb", "ccc"));
+    record.put(COMPLEX_LIST,
+        Arrays.asList(
+            createNestedMap(NESTED_STRING_FIELD, "hows", NESTED_INT_FIELD, 10),
+            createNestedMap(NESTED_STRING_FIELD, "it", NESTED_INT_FIELD, 20),
+            createNestedMap(NESTED_STRING_FIELD, "going", NESTED_INT_FIELD, 30)
+        )
+    );
+    record.put(SIMPLE_MAP, createNestedMap("Tuesday", 3, "Wednesday", 4));
+    record.put(
+        COMPLEX_MAP,
+        createNestedMap(
+            "fruit1", createNestedMap(NESTED_STRING_FIELD, "apple", NESTED_INT_FIELD, 1),
+            "fruit2", createNestedMap(NESTED_STRING_FIELD, "orange", NESTED_INT_FIELD, 2)
+        )
+    );
+    return record;
+  }
+
+  private Map<String, Object> createNestedMap(String key1, Object value1, String key2, Object value2) {
+    Map<String, Object> nestedMap = new HashMap<>(2);
+    nestedMap.put(key1, value1);
+    nestedMap.put(key2, value2);
+    return nestedMap;
+  }
+
+  private NestedType createNestedType(String stringField, int intField) {
+    NestedType nestedRecord = new NestedType();
+    nestedRecord.setNestedStringField(stringField);
+    nestedRecord.setNestedIntField(intField);
+    return nestedRecord;
+  }
+
+  @Override
+  protected boolean testExtractAll() {
```

Comment:
if this is true for all extractor tests now, can we remove this method and just default to testing extractAll?

Reply:
Good call. We can remove this method now and testing the `extract all` code path will be part of all the tests.